instrumentation/graphql-java/graphql-java-20.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/graphql/v20_0/GraphqlSingletons.java

```diff
@@ -5,44 +5,97 @@
-      GraphQLTelemetry.builder(GlobalOpenTelemetry.get())
-          .setCaptureQuery(CAPTURE_QUERY)
-          .setSanitizeQuery(QUERY_SANITIZATION_ENABLED)
-          .setDataFetcherInstrumentationEnabled(DATA_FETCHER_ENABLED)
-          .setTrivialDataFetcherInstrumentationEnabled(TRIVIAL_DATA_FETCHER_ENABLED)
-          .setAddOperationNameToSpanName(ADD_OPERATION_NAME_TO_SPAN_NAME)
-          .build();
+  private static final GraphQLTelemetry TELEMETRY;
+
+  static {
+    Configuration config = new Configuration(GlobalOpenTelemetry.get());
+
+    TELEMETRY =
+        GraphQLTelemetry.builder(GlobalOpenTelemetry.get())
+            .setCaptureQuery(config.captureQuery(true))
+            .setSanitizeQuery(config.sanitizeQuery(true))
+            .setDataFetcherInstrumentationEnabled(config.dataFetcherEnabled(false))
+            .setTrivialDataFetcherInstrumentationEnabled(config.trivialDataFetcherEnabled(false))
+            .setAddOperationNameToSpanName(config.addOperationNameToSpanName(false))
+            .build();
+  }
 
   private GraphqlSingletons() {}
 
   public static Instrumentation addInstrumentation(Instrumentation instrumentation) {
     Instrumentation ourInstrumentation = TELEMETRY.newInstrumentation();
     return InstrumentationUtil.addInstrumentation(instrumentation, ourInstrumentation);
   }
+
+  // instrumentation/development:
+  //   java:
+  //     graphql:
+  //       capture_query: [boolean]
+  //       query_sanitizer:
+  //         enabled: [boolean]
+  //       data_fetcher:
+  //         enabled: [boolean]
+  //       trivial_data_fetcher:
+  //         enabled: [boolean]
+  //       add_operation_name_to_span_name:
+  //         enabled: [boolean]
+  private static final class Configuration {
+
+    @Nullable private final Boolean captureQuery;
+    @Nullable private final Boolean sanitizeQuery;
+    @Nullable private final Boolean dataFetcherEnabled;
+    @Nullable private final Boolean trivialDataFetcherEnabled;
+    @Nullable private final Boolean addOperationNameToSpanName;
+
+    Configuration(OpenTelemetry openTelemetry) {
+      DeclarativeConfigProperties graphqlConfig =
+          DeclarativeConfigUtil.getStructured(openTelemetry, "java", empty())
+              .getStructured("graphql", empty());
+
+      this.captureQuery = graphqlConfig.getBoolean("capture_query");
+      this.sanitizeQuery =
+          graphqlConfig.getStructured("query_sanitizer", empty()).getBoolean("enabled");
+      this.dataFetcherEnabled =
+          graphqlConfig.getStructured("data_fetcher", empty()).getBoolean("enabled");
+      this.trivialDataFetcherEnabled =
+          graphqlConfig.getStructured("trivial_data_fetcher", empty()).getBoolean("enabled");
+      this.addOperationNameToSpanName =
+          graphqlConfig
+              .getStructured("add_operation_name_to_span_name", empty())
+              .getBoolean("enabled");
+    }
+
+    boolean captureQuery(boolean defaultValue) {
```

Comment:
Do you like have the default value be defined here? You could initialize each of the properties on lines 57-61 with default values using the overloads in `DeclarativeConfigProperties` like:

```
this.captureQuery = graphqlConfig.getBoolean("capture_query", true)
```

Benefit is that the properties don't have to be nullable. Downside is you don't get to see the default values laid out in lines 27-33.

Reply:
yeah, I think I'm leaning towards moving the defaults here, given that `defaultBehavior` is now part of the schema https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/15641#discussion_r2620844181